src/librustc/middle/weak_lang_items.rs

```diff
@@ -1,49 +1,25 @@
 //! Validity checking for weak lang items
 
 use crate::middle::lang_items;
-use crate::session::config;
 
-use crate::hir;
 use crate::hir::def_id::DefId;
-use crate::hir::intravisit;
-use crate::hir::intravisit::{NestedVisitorMap, Visitor};
 use crate::ty::TyCtxt;
-use rustc_data_structures::fx::FxHashSet;
+use rustc_data_structures::fx::FxHashMap;
 use rustc_target::spec::PanicStrategy;
 use syntax::ast;
 use syntax::symbol::{sym, Symbol};
-use syntax_pos::Span;
 
-use rustc_error_codes::*;
+use lazy_static::lazy_static;
 
 macro_rules! weak_lang_items {
```

Comment:
Any particular reason you where unable to move the entire file?

Reply:
`librustc_codegen_ssa` depends on the `link_name` function. I did not want to introduce a dependency on `librustc_passes` just for that.

Reply:
I'd just add that dependency instead of keeping stuff in rustc.

Reply:
Please, don't introduce the dependency, it's a much larger build time penalty than a tiny function in rustc.

Reply:
That is not true though.

Reply:
Maybe make a `rustc_lang_items` and put that as a `librustc_codegen_ssa` dependency instead?

Reply:
`rustc_lang_items` is probably sensible long term, also because I'd like to be able to refer to lang items in lowering (https://github.com/rust-lang/rust/issues/60607).

Reply:
Let's just move these to a new `rustc_lang_items` crate then.